x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/openai/embeddings/OpenAiEmbeddingsServiceSettings.java

```diff
@@ -58,22 +57,19 @@ public static OpenAiEmbeddingsServiceSettings fromMap(Map<String, Object> map, C
     }
 
     private static OpenAiEmbeddingsServiceSettings fromPersistentMap(Map<String, Object> map) {
+        // Reading previously persisted config, assume the validation
+        // passed at that time and never throw.
         ValidationException validationException = new ValidationException();
 
         var commonFields = fromMap(map, validationException);
 
         Boolean dimensionsSetByUser = removeAsType(map, DIMENSIONS_SET_BY_USER, Boolean.class);
         if (dimensionsSetByUser == null) {
-            validationException.addValidationError(
```

Comment:
Welp I guess that proves how easy it is to forget that all new settings must be optional and your point about validation haha, sorry about that 😞.

We should remove the validation for all the other services when reading from the persistent config. I can work on that.

Also we should backport this particular fix to when the setting was introduced too. Otherwise when users upgrade they'll have to delete the inference endpoint and re-add them I think. I think it was introduced in 8.13.